support-services/src/main/scala/com/gu/support/redemption/DynamoTableAsync.scala

```diff
@@ -0,0 +1,86 @@
+package com.gu.support.redemption
+
+import java.util.concurrent.CompletionException
+
+import com.typesafe.scalalogging.LazyLogging
+import software.amazon.awssdk.auth.credentials.{AwsCredentialsProviderChain, InstanceProfileCredentialsProvider, ProfileCredentialsProvider}
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, GetItemRequest, UpdateItemRequest}
+
+import scala.collection.JavaConverters._
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+object DynamoTableAsync {
+
+  val ProfileName = "membership"
+
+  lazy val CredentialsProvider =  AwsCredentialsProviderChain.builder.credentialsProviders(
+     ProfileCredentialsProvider.builder.profileName(ProfileName).build,
+     InstanceProfileCredentialsProvider.builder.asyncCredentialUpdateEnabled(false).build
+  ).build
+
+  def apply(table: String, key: String)(implicit e: ExecutionContext): DynamoTableAsync = {
+    val dynamoDBClient = DynamoDbAsyncClient.builder
+      .credentialsProvider(CredentialsProvider)
+      .region(Region.EU_WEST_1)
+      .build
+    new DynamoTableAsync(dynamoDBClient, table, key)
+  }
+}
+trait DynamoLookup {
+  def lookup(value: String): Future[Option[Map[String, Boolean]]]
+}
+
+trait DynamoUpdate {
+  def update(primaryKeyValue: String, updateName: String, updateValue: Boolean): Future[Unit]
+}
+
+class DynamoTableAsync(
+  dynamoDBAsyncClient: DynamoDbAsyncClient,
+  table: String,
+  primaryKeyName: String
+)(implicit e: ExecutionContext) extends LazyLogging with DynamoLookup with DynamoUpdate {
+
+  override def lookup(primaryKeyValue: String): Future[Option[Map[String, Boolean]]] = {
```

Comment:
is `primaryKeyValue` the redemptionCode? If so it might be be good to call this parameter `redemptionCode`? Less mental translation to do...

Reply:
it depends what we passed in as primaryKeyName on line 43.
I think the decision here is hinged on whether this is a redemptions client or a dynamo-to-scala layer as per the above discussion.